src/rpm_ostree/cli_status.rs

```diff
@@ -143,24 +144,23 @@ fn booted_json(status: &StatusJSON) -> Fallible<DeploymentJSON> {
     Ok(booted)
 }
 
-/// Introspect deployments (rpm-ostree status) using rpm-ostree client actor client's
-/// cache if possible.
-fn status_json(client: &mut RpmOstreeClient) -> Fallible<StatusJSON> {
+/// Ensure our status cache is up to date; if empty or out of date, run `rpm-ostree status` to populate it.
+fn status_json(client: &mut RpmOstreeClient) -> Fallible<Rc<StatusJSON>> {
```

Comment:
I think it's really neat how in Rust one can change a function like this to return an `Rc<>` and for all the callers it's transparent due to the auto-deref.

Reply:
It took some getting used to. I remember earlier on when I almost forgot that `&` are just references instead of a magical symbol used for borrowing stuff.